### PR TITLE
Remove Obsolete ShardGenerations BwC Logic

### PR DESCRIPTION
**Pull request by original-brownbear**
With #48371 merged there is no need to keep the BwC logic
for writing snapshots in the old style.
The only thing necessary to retain is the `SnapshotsInProgress.Entry`
serialization BwC.

### CODE REVIEW

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/bwc  (failed to build only)

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/default-distro (random build stuff)

**Comment by original-brownbear** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java [hunk 1]
```diff
@@ -520,11 +505,9 @@ public SnapshotsInProgress(StreamInput in) throws IOException {
             if (in.getVersion().onOrAfter(METADATA_FIELD_INTRODUCED)) {
                 userMetadata = in.readMap();
             }
-            final boolean useShardGenerations;
-            if (in.getVersion().onOrAfter(SnapshotsService.SHARD_GEN_IN_REPO_DATA_VERSION)) {
-                useShardGenerations = in.readBoolean();
-            } else {
-                useShardGenerations = false;
+            final Version version = in.getVersion();
+            if (version.before(Version.V_8_0_0) && version.onOrAfter(SnapshotsService.SHARD_GEN_IN_REPO_DATA_VERSION)) {
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java]: given that this logic can only be removed in 9.0, perhaps simpler to just unconditionally do a  `in.readBoolean();` here and add an assertion that fails once we branch 9.0. Also note that `version.onOrAfter(SnapshotsService.SHARD_GEN_IN_REPO_DATA_VERSION)` should be unnecessary once we stop testing master against 7.5. This is also why I prefer to wait with PRs like this until that has happened, it simplifies the BWC conditions.

Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java]: >  This is also why I prefer to wait with PRs like this until that has happened, it simplifies the BWC conditions.

Right that's a fair point. I'll just close here then and look into this again once `7.5` is out :)

**Pull request closed without merging**